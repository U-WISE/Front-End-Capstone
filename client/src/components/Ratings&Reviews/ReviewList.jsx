import React from 'react';
import Review from './Review.jsx';

const ReviewList = () => (
  <div id="reviewList">
    <div>Number of Reviews, Sorted By Sort</div>
    <Review />
    <button>More Reviews</button>
    <button>Add a review +</button>
  </div>
);
export default ReviewList;
