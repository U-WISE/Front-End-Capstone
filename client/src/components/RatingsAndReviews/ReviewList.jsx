import React, { useState, useContext } from 'react';
import Review from './Review.jsx';
import { reviewsContext } from '../../contexts/index.js';
import { toggleModal } from './utils/RatingsAndReviews.utils.js';

const ReviewList = () => {
  const context = useContext(reviewsContext);
  const [loadedReviews, setLoadedReviews] = useState(2);

  const moreReviews = () => {
    if (context.reviews.length > 2 && loadedReviews < context.reviews.length) {
      return <button onClick={() => setLoadedReviews(loadedReviews + 2)}>More Reviews</button>;
    }
    return null;
  };

  const anyReviews = () => {
    if (context.reviews.length > 0) {
      return <div> {context.reviews.length} reviews, Sorted By Sort</div>;
    }
    return null;
  };

  return (
    <div id="reviewList">
      {anyReviews()}
      <div id="reviews">
        {context.reviews.slice(0, loadedReviews).map((review) => (
          <Review
            key={review.review_id}
            review={review}
          />
        ))}
      </div>
      {moreReviews()}
      <button
        onClick={() => { toggleModal(context.modalVisibility, context.setModalVisibility); }}
      >Add a review +
      </button>
    </div>
  );
};
export default ReviewList;
