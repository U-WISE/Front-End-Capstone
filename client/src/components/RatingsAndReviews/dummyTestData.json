{
  "reviews": [
    {
      "review_id": 720891,
      "rating": 5,
      "summary": "This product was great!",
      "recommend": true,
      "response": "",
      "body": "I really did or did not like this product based on whether it was sustainably sourced. Then I found out that its made from nothing at all.",
      "date": "2019-01-01T00:00:00.000Z",
      "reviewer_name": "funtime",
      "helpfulness": 8,
      "photos": []
    },
    {
      "review_id": 720892,
      "rating": 4,
      "summary": "This product was ok!",
      "recommend": false,
      "response": "",
      "body": "I really did not like this product solely because I am tiny and do not fit into it.",
      "date": "2019-01-11T00:00:00.000Z",
      "reviewer_name": "mymainstreammother",
      "helpfulness": 2,
      "photos": []
    }
  ],
  "reviewsMeta": {
    "product_id": "44388",
    "ratings": {
      "4": "1",
      "5": "1"
    },
    "recommended": {
      "false": "1",
      "true": "1"
    },
    "characteristics": {
      "Fit": {
        "id": 148890,
        "value": "4.0000000000000000"
      },
      "Length": {
        "id": 148891,
        "value": "3.5000000000000000"
      },
      "Comfort": {
        "id": 148892,
        "value": "5.0000000000000000"
      },
      "Quality": {
        "id": 148893,
        "value": "4.0000000000000000"
      }
    }
  },
  "filtered": {
    "one": false,
    "two": false,
    "three": false,
    "four": false,
    "five": false
  }
}