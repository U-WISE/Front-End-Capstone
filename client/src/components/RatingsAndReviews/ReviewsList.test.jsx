import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import ReviewList from './ReviewList.jsx';
import { reviewsContext } from '../../contexts/index.js';
import { reviews } from './dummyTestData.json';

test('to be defined"', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews }}>
      <ReviewList />
    </reviewsContext.Provider>
  );
  expect(component).toBeDefined();
});

test('to undefined', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews }}>
      <ReviewList />
    </reviewsContext.Provider>
  );
  expect(component.anUndefinedValue).toBeUndefined();
});

test('to match snapshot', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews }}>
      <ReviewList />
    </reviewsContext.Provider>
  );
  expect(component).toMatchSnapshot();
});