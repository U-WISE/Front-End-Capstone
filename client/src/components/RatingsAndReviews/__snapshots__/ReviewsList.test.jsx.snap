// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  id="reviewList"
>
  <div>
     
    2
     reviews, Sorted By Sort
  </div>
  <div
    id="reviews"
  >
    <div
      id="ReviewContainer"
    >
      <div
        id="starRating"
      >
        StarRating
      </div>
      <div
        id="verifiedPurchaser"
      >
        Verified Purchaser
      </div>
      <span
        id="reviewUserName"
      >
        funtime
      </span>
      <span
        id="reviewDate"
      >
        January 01, 2019
      </span>
      <span
        id="reviewSummary"
      >
        This product was great!
      </span>
      <p
        id="reviewBody"
      >
        I really did or did not like this product based on whether it was sustainably sourced. Then I found out that its made from nothing at all.
      </p>
      <span
        id="reviewHelpful"
      >
        Helpful?
        <button
          className="helpfulnessLinks"
          id="reviewYes"
        >
          Yes(
          8
          )
        </button>
        <button
          className="helpfulnessLinks"
          id="reviewNo"
        >
          No(
          8
          )
        </button>
      </span>
      <button
        id="reviewReport"
      >
        Report
      </button>
    </div>
    <div
      id="ReviewContainer"
    >
      <div
        id="starRating"
      >
        StarRating
      </div>
      <div
        id="verifiedPurchaser"
      >
        Verified Purchaser
      </div>
      <span
        id="reviewUserName"
      >
        mymainstreammother
      </span>
      <span
        id="reviewDate"
      >
        January 11, 2019
      </span>
      <span
        id="reviewSummary"
      >
        This product was ok!
      </span>
      <p
        id="reviewBody"
      >
        I really did not like this product solely because I am tiny and do not fit into it.
      </p>
      <span
        id="reviewHelpful"
      >
        Helpful?
        <button
          className="helpfulnessLinks"
          id="reviewYes"
        >
          Yes(
          2
          )
        </button>
        <button
          className="helpfulnessLinks"
          id="reviewNo"
        >
          No(
          2
          )
        </button>
      </span>
      <button
        id="reviewReport"
      >
        Report
      </button>
    </div>
  </div>
  <button>
    Add a review +
  </button>
</div>
`;
